e2e-tests/src/protocol/evm.rs

```diff
@@ -0,0 +1,69 @@
+use core::time::Duration;
+use std::net::TcpStream;
+
+use eyre::{bail, OptionExt, Result as EyreResult};
+use serde::{Deserialize, Serialize};
+use url::Url;
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct EvmProtocolConfig {
+    pub context_config_contract_id: String,
+    pub rpc_url: String,
+    pub account_id: String,
+    pub secret_key: String,
+}
+
+pub struct EvmSandboxEnvironment {
+    config: EvmProtocolConfig,
+}
+
+impl EvmSandboxEnvironment {
+    pub fn init(config: EvmProtocolConfig) -> EyreResult<Self> {
+        let rpc_url = Url::parse(&config.rpc_url)?;
+        let rpc_host = rpc_url
+            .host_str()
+            .ok_or_eyre("failed to get evm rpc host from config")?;
+        let rpc_port = rpc_url
+            .port()
+            .ok_or_eyre("failed to get evm rpc port from config")?;
+
+        if let Err(err) = TcpStream::connect_timeout(
+            &format!("{rpc_host}:{rpc_port}").parse()?,
+            Duration::from_secs(3),
+        ) {
+            bail!(
+                "Failed to connect to evm rpc url '{}': {}",
+                &config.rpc_url,
+                err
+            );
+        }
+
+        Ok(Self { config })
+    }
+
+    pub fn node_args(&self) -> Vec<String> {
+        println!("config: {:?}", self.config);
+        vec![
+            format!("context.config.evm.protocol=\"{}\"", "evm"),
```

Comment:
why am I just realizing we have a duplication with this

```toml
[context.config.near]
signer = "relayer"
protocol = "near"
network = "testnet"
contract_id = "calimero-context-config.testnet"
```

the `protocol` field is redundant (already named in the table), we can fix in a separate PR

Reply:
Filed in #1149.